### PR TITLE
update fxcop to version 2.9.8

### PR DESCRIPTION
**Pull request by TimothyMothra**

### CODE REVIEW

**Comment by TimothyMothra** [comment]
@Dmitry-Matveev I need your review. If this is too large, please let me know and I can split this into smaller PRs

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 1]
```diff
@@ -53,6 +53,24 @@
     </PackageReference>
   </ItemGroup>
 
+  <Choose>
+    <When Condition=" '$(Configuration)' == 'Release' ">
+      <PropertyGroup>
+        <!-- Microsoft recommends that we disable CodeAnalysis in favor of FxCop Analyzers. aka.ms/fxcopanalyzers -->
+        <RunCodeAnalysis>false</RunCodeAnalysis>
+        <CodeAnalysisRuleSet>$(RulesetsRoot)\ApplicationInsightsSDKRules.ruleset</CodeAnalysisRuleSet>
```
Comment (Dmitry-Matveev) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj]: Is this ruleset only for certain SDKs? Other projects above didn't have it.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs [hunk 2]
```diff
@@ -65,6 +65,21 @@ public string HeaderValueSeparators
...
         protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
         {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (platformContext == null)
+            {
+                throw new ArgumentNullException(nameof(platformContext));
```
Comment (Dmitry-Matveev) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs]: This now throws if even if requestTelemetry.Context.Location.Ip is not empty. We need to check if upstream code is Ok with this.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs [hunk 3]
```diff
@@ -21,6 +22,21 @@ public OperationNameTelemetryInitializer(IHttpContextAccessor httpContextAccesso
...
         protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
         {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (platformContext == null)
+            {
+                throw new ArgumentNullException(nameof(platformContext));
```
Comment (Dmitry-Matveev) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs]: This now throws even if requestTelemetry.name is not empty and execution does not reach platformContext branch.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/SyntheticTelemetryInitializer.cs [hunk 4]
```diff
@@ -26,6 +27,16 @@ public SyntheticTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
...
         protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
         {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (platformContext == null)
+            {
+                throw new ArgumentNullException(nameof(platformContext));
```
Comment (Dmitry-Matveev) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/SyntheticTelemetryInitializer.cs]: Similar here and with other checks on platformContext.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj (hunk 1 above)
Comment (TimothyMothra) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj]: I merged the rulesets. This is now a superset of all repos.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/MvcDiagnosticsListener.cs [hunk 5]
```diff
@@ -31,6 +32,16 @@ public class MvcDiagnosticsListener : IApplicationInsightDiagnosticListener
...
         public void OnBeforeAction(HttpContext httpContext, IDictionary<string, object> routeValues)
         {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            if (routeValues == null)
+            {
+                throw new ArgumentNullException(nameof(routeValues));
```
[the remark this answers is not in the document]
Comment (TimothyMothra) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/MvcDiagnosticsListener.cs]: Spoke offline. I'm going to move the null check into the if statement.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs (hunk 2 above)
Comment (TimothyMothra) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs]: moved into if

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs (hunk 3 above)
Comment (TimothyMothra) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs]: moved into if

**Pull request merged**